src/System.Runtime.Numerics/src/System/Numerics/Complex.cs

```diff
@@ -115,11 +115,17 @@ public static double Abs(Complex value)
                 return double.PositiveInfinity;
             }
 
+            return Hypot(value._real, value._imaginary);
+        }
+
+        private static double Hypot (double a, double b)
+        {
             // |value| == sqrt(a^2 + b^2)
```

Comment:
Because PositiveInfinity or NegativeInfinity could be passed in, as a component of some complex whose Abs or Sqrt we are trying to compute. And this Hypot behaves as it should when that happens, returning PositiveInfinity.